src/control/settings/Settings.cpp

```diff
@@ -464,11 +469,32 @@ void Settings::loadDeviceClasses() {
     }
 }
 
+const char* getButtonConfigName(int button) {
+    switch (button) {
+        case BUTTON_ERASER:
+            return "eraser";
+        case BUTTON_MIDDLE:
+            return "middle";
+        case BUTTON_RIGHT:
+            return "right";
+        case BUTTON_TOUCH:
+            return "touch";
+        case BUTTON_DEFAULT:
+            return "default";
+        case BUTTON_STYLUS:
+            return "stylus";
+        case BUTTON_STYLUS2:
+            return "stylus2";
+        default:
+            return "unknown";
+    }
+}
```

Comment:
Make this function constexpr.
Note: Also the best would be to return string_views.

Reply:
I am a bit rusty with C++ but will give it a try

Reply:
The constexpr is fine but I get an error with string_view because string_views are not necessarily null-terminated while the function `s.child()` that takes the output of `getButtonConfigName` needs something that can be converted to `string` (its signature is `SElement::child(const string& name)`). On the other hand `string` cannot be `constexpr` in C++17. The two options I can see are `constexpr const char*` or `string` (like, e.g., in `drawingTypeToString` in ToolEnums.cpp). Any preference? I've gone with the first one for now but happy to change if you prefer.
PS: Also, I have now used `auto` with a trailing data type as this seems to be the style used in the rest of the file.

Reply:
Yes I knew about that habbit. Thats why I marked it as note. I prefer a char const* as return value. But if you still want to return a string_view you can still convert it into a string. A string can also be constructed from a char array without a null termination by providing the size of the char array.